src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -389,7 +398,8 @@ private void signalIfAllRegisteredInQueue() {
         // is active at a time - a caller of lock() that has just acquired
         // the pool lock but is waiting for all registered elements to be
         // returned to the queue.
-        if (registeredElements.size() == liveQueue.size()) {
+        if (registeredElements.size() == liveQueue.size() ||
+                poisonPill != null) {
```

Comment:
I've gone back and looked at what uses `allRegisteredInQueue.await()`, and it looks like it's only used in `.lock`. So I think there's no reason =)

I think I misunderstood what it was used for when I made the change, so thanks for pointing this out